src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebRequestPSCmdlet.Common.cs

```diff
@@ -173,6 +173,12 @@ public virtual string CustomMethod
         #region Proxy
 
         /// <summary>
+        /// gets or sets the NoProxy property
+        /// </summary>
+        [Parameter]
+        public virtual SwitchParameter NoProxy { get; set; }
```

Comment:
For consistency, it seems like these should be mutually exclusive (thus different parametersets).  PowerShell doesn't work like Unix cmdlines where last one wins (in which case, the above example would be -Proxy overrides -NoProxy despite $true or $false).  I've typically used splatting to construct parameters dynamically.  I think if we want something more terse, we should introduce a more general approach as a language construct.